workers/shared-worker-options-mismatch.html

```diff
@@ -0,0 +1,90 @@
+<!DOCTYPE html>
+<title>SharedWorker: type or credentials mismatch failure</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script> setup({allow_uncaught_exception: true}); </script>
+<script>
+
+const mismatch_test = (testName, firstOptions, secondOptions) => {
+  promise_test(async () => {
+    firstOptions.name = testName;
+    secondOptions.name = testName;
+    const scriptURL = 'support/empty-worker.js';
+    const firstWorker = new SharedWorker(scriptURL, firstOptions);
+    const secondWorker = new SharedWorker(scriptURL, secondOptions);
+    return new Promise(resolve => secondWorker.onerror = resolve);
+  }, 'Connecting to shared worker with different options should be blocked: '
+  + testName);
+};
+
+// Tests different type.
+mismatch_test('default to module', {}, { type: 'module' });
+mismatch_test('module to default', { type: 'module' }, {});
+mismatch_test('classic to module', { type: 'classic' }, { type: 'module' });
+mismatch_test('module to classic', { type: 'module' }, { type: 'classic' });
+
+// Tests different credentials in classic.
+mismatch_test('default to omit in classic',
+              { type: 'classic' },
+              { type: 'classic', credentials: 'omit' });
+mismatch_test('default to include in classic',
+              { type: 'classic' },
+              { type: 'classic', credentials: 'include' });
+mismatch_test('omit to default in classic',
+              { type: 'classic', credentials: 'omit' },
+              { type: 'classic' });
+mismatch_test('omit to same-origin in classic',
+              { type: 'classic', credentials: 'omit' },
+              { type: 'classic', credentials: 'same-origin' });
+mismatch_test('omit to include in classic',
+              { type: 'classic', credentials: 'omit' },
+              { type: 'classic', credentials: 'include' });
+mismatch_test('same-origin to omit in classic',
+              { type: 'classic', credentials: 'same-origin'},
+              { type: 'classic', credentials: 'omit' });
+mismatch_test('same-origin to include in classic',
+              { type: 'classic', credentials: 'same-origin'},
+              { type: 'classic', credentials: 'include' });
+mismatch_test('include to default in classic',
+              { type: 'classic', credentials: 'include' },
+              { type: 'classic' });
+mismatch_test('include to omit in classic',
+              { type: 'classic', credentials: 'include' },
+              { type: 'classic', credentials: 'omit' });
+mismatch_test('include to same-origin in classic',
+              { type: 'classic', credentials: 'include' },
+              { type: 'classic', credentials: 'same-origin' });
+
+// Tests different credentials in module.
```

Comment:
Are these essentially the same as classic? I'd prefer a `["classic", "module"].forEach` wrapper in that case to make that clear.

Reply:
As you said, the tests for 'module' is same to that of 'classic', so I modified to use forEach wrapper.